alb/main.tf

```diff
@@ -0,0 +1,84 @@
+locals {
+  lb_name = "${var.environment}-${var.name}"
+}
+
+resource "aws_security_group" "alb" {
+  name        = local.lb_name
+  description = "${local.lb_name} ALB default SG"
+  vpc_id      = var.vpc_id
+}
+
+resource "aws_security_group_rule" "egress" {
+  security_group_id = aws_security_group.alb.id
+
+  type        = "egress"
+  from_port   = "0"
+  to_port     = "0"
+  protocol    = "-1"
+  cidr_blocks = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "ingress_http" {
+  security_group_id = aws_security_group.alb.id
+
+  type        = "ingress"
+  from_port   = 80
+  to_port     = 80
+  protocol    = "tcp"
+  cidr_blocks = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "ingress_https" {
+  security_group_id = aws_security_group.alb.id
+
+  type        = "ingress"
+  from_port   = 443
+  to_port     = 443
+  protocol    = "tcp"
+  cidr_blocks = ["0.0.0.0/0"]
+}
+
+# load balancer
+resource "aws_lb" "this" {
+  name                       = local.lb_name
+  internal                   = false
+  load_balancer_type         = "application"
+  security_groups            = [aws_security_group.alb.id]
+  subnets                    = var.subnet_ids
+  enable_deletion_protection = true
+  ip_address_type            = "ipv4"
+}
+
+# listeners
+resource "aws_lb_listener" "http" {
+  load_balancer_arn = aws_lb.this.arn
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_lb_listener" "https" {
+  load_balancer_arn = aws_lb.this.arn
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-2016-08"
+  certificate_arn   = var.certificate_arn
+
+  default_action {
+    type = "fixed-response"
+
+    fixed_response {
+      content_type = "text/html"
+      status_code  = "404"
```

Comment:
ezt elmagyarazhatnad nekem h itt mit tortenik, esetleg mehet comment-be is

Reply:
ez a default action a ruleokban, ha nincs olyan rule ami raillene egy requestre, akkor ezt adjuk vissza. 404 sztem defaultnak jo.

Lehetne ez defaultban lambdahoz iranyit, v egy target grouphoz, vagy mint fentebb, defaultban 301 HTTPS-re redirectel.

Reply:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_listener#default_action

Reply:
vagy egy jo defaiult page-re mint az ingress-nginx default backend... de jo ez igy most sztem, koszi

Reply:
jah, siman betehetunk ide html-t is es 200-at kesobb